SimulationAgent/Agent.cs

```diff
@@ -231,6 +237,23 @@ private void StopInactiveSimulations(IList<Simulation> activeSimulations)
             }
         }
 
+        private async Task SaveSimulationStatisticsAsync(IList<Simulation> simulations)
+        {
+            foreach (var simulation in simulations)
+            {
+                DateTimeOffset now = DateTimeOffset.UtcNow;
+                TimeSpan duration = now - this.lastPolledTime;
+
+                // Save simulation statistics at specified interval
+                if (duration.Seconds >= STATS_INTERVAL_SECS)
+                {
+                    this.simulationManagers[simulation.Id].SaveStatisticsAsync();
```

Comment:
same here, is `await` missing on purpose...etc...

Reply:
does this code work? I would expect to throw an exception because the simulation manager is created after invoking this method the first time (see the order of lines 185 and 186)

Reply:
added await.

good catch about the order, the code does work but I had not tried with STATS_INTERVAL_SECS duration too high or low. updated the order.